src/Aquifer.Common/Services/ResourceContentSearchService.cs

```diff
@@ -0,0 +1,672 @@
+                        WHERE
+                            r.Id = psr.ResourceId AND
+                            (
+                                p.StartVerseId BETWEEN @{startVerseIdParamName} AND @{endVerseIdParamName} OR
+                                p.EndVerseId BETWEEN @{startVerseIdParamName} AND @{endVerseIdParamName} OR
+                                (p.StartVerseId <= @{startVerseIdParamName} AND p.EndVerseId >= @{endVerseIdParamName})
+                            )
+                    )
+                )
+            """);
+        }
+
+        if (filter.AssignedUserId.HasValue)
+        {
+            const string assignedUserIdParamName = "assignedUserId";
+
+            coreParameters.Add(assignedUserIdParamName, filter.AssignedUserId.Value);
+            whereClausesSql.Add($"rcv.AssignedUserId = @{assignedUserIdParamName}");
+        }
+
+        if (filter.AssignedUserCompanyId.HasValue)
+        {
+            const string assignedUserCompanyIdParamName = "assignedUserCompanyId";
+
+            coreParameters.Add(assignedUserCompanyIdParamName, filter.AssignedUserCompanyId.Value);
+            whereClausesSql.Add($"u.CompanyId = @{assignedUserCompanyIdParamName}");
+        }
+
+        if (filter.IsPublished.HasValue)
+        {
+            if (filter.IsPublished.Value)
+            {
+                whereClausesSql.Add(includeFlags.HasFlag(ResourceContentSearchIncludeFlags.ResourceContentVersions)
+                    ? "rcvd.AnyIsPublished = 1"
+                    : "rcv.IsPublished = 1");
+            }
+            else
+            {
+                // supporting this search will add complexity and should only be added if needed
+                throw new NotImplementedException($"A \"{nameof(filter.IsPublished)}\" value of false is not yet supported.");
+            }
+        }
+
+        if (filter.IsDraft.HasValue)
+        {
+            if (filter.IsDraft.Value)
+            {
+                whereClausesSql.Add(includeFlags.HasFlag(ResourceContentSearchIncludeFlags.ResourceContentVersions)
+                    ? "rcvd.AnyIsDraft = 1"
+                    : "rcv.IsDraft = 1");
+            }
+            else
+            {
+                // supporting this search will add complexity and should only be added if needed
+                throw new NotImplementedException($"A \"{nameof(filter.IsDraft)}\" value of false is not yet supported.");
+            }
+        }
+
+        var whereSql = whereClausesSql.Count > 0
+            ? $"WHERE{Environment.NewLine}    {string.Join($" AND{Environment.NewLine}    ", whereClausesSql)}"
+            : "";
+
+        var hasPaging = offset != 0 || limit != null;
+
+        const string offsetLiteralName = "offset";
+        const string limitLiteralName = "limit";
+        const string pagingSql = $$"""
+           OFFSET {={{offsetLiteralName}}} ROWS FETCH NEXT {={{limitLiteralName}}} ROWS ONLY
```

Comment:
Dapper allows injecting literals.  This won't use SQL parameters but still allows populating the parameters object with the literal values.  I like this methodology because it makes it clear that you aren't using SQL parameters on purpose and you know there isn't a risk of SQL injection attacks (Dapper only supports safe values for literals).